sklearn/tests/test_dummy.py

```diff
@@ -571,6 +571,33 @@ def test_classification_sample_weight():
     assert_array_almost_equal(clf.class_prior_, [0.2 / 1.2, 1. / 1.2])
 
 
+def test_sample_weight_invalid():
+    # Check sample weighting raises errors.
+    X = [[0], [0], [1]]
+    y = [0, 1, 0]
+    sample_weight = [0.1, 1., 0.1]
+
+    clf = DummyClassifier().fit(X, y, sample_weight)
+    assert_array_almost_equal(clf.class_prior_, [0.2 / 1.2, 1. / 1.2])
+
+    sample_weight = np.random.rand(3, 1)
```

Comment:
For anything below this line, I'm not sure there is much sense in adding sample_weight tests for each estimator (unless they have some special behavior there) as they would be quite redundant, instead they should be enforced by common tests in `sklearn/utils/estimator_checks.py` (and for a number of below point they may already be).

So I would remove the below test. The check specific to `DummyClassifier` above is quite nice though.

Reply:
Thanks for the review! 

We removed the test below.